### ISSUE TITLE
`no-nested-object-setvalue` throws when passing a function call

### ISSUE THREAD

**Issue by akd-io** [andykao1213/eslint-plugin-react-hook-form#17]
**Describe the bug**
`react-hook-form/no-nested-object-setvalue` is thrown with message `Avoid passing object or array as second argument in setValue since this is less performant`, when using the following syntax:

```typescript
setValue("someName", myFunctionReturningANumber())
```

**To Reproduce**
1. The code you are going to lint.
```tsx
function Component() {
  const {useValue} = useForm();

  const myFunctionReturningANumber = () => 5

  setValue("someName", myFunctionReturningANumber())

  return null;
}
```
2. Your ESlint config
```
{
  ...
  'react-hook-form/no-nested-object-setvalue': ['error', { bracketAsArrayIndex: true }],
}
```

**Expected behavior**
Expect `react-hook-form/no-nested-object-setvalue` not to throw an error.

**Actual behavior**
`react-hook-form/no-nested-object-setvalue` throws with message `Avoid passing object or array as second argument in setValue since this is less performant`

**Enviornment (please complete the following information):**
- Eslint version: 7.12.1
- Node version: v14.18.3
- React-Hook-Form version: 6.15.8
- eslint-plugin-react-hook-form version: 0.2.3

**Additional context**
n/a

**Comment by andykao1213** [comment]
@akd-io Thanks for opening the issue. Currently, the rule will throw an error if the second argument of the `setValue` is not a [Literal](https://github.com/estree/estree/blob/master/es5.md#literal). This should also throw an error when we pass a variable to the `setValue`, which is not ideal.
To solve this problem, my proposal is only to throw errors if we directly pass arguments with the `ObejctExpression` or `ArrayExpression`. Other than those should be fine. But one should notice that if passing variable or execute function will not work:
```js
const { setValue } = useForm()
// These will throw an error
setValue('field1', [....]) 
setValue('field2', {...})
// These won't throw error
const foo = () => []
setValue('field3', foo())
const arr = []
setValue('field3', arr)
```
Let me know if you have any concerns about this. Thank you.

**Comment by andykao1213** [comment]
Issue fixed in [v0.2.4](https://github.com/andykao1213/eslint-plugin-react-hook-form/releases/tag/v0.2.4)
